python/az/aro/azext_aro/_rbac.py

```diff
@@ -51,19 +49,44 @@ def assign_contributor_to_vnet(cli_ctx, vnet, object_id):
         name=NETWORK_CONTRIBUTOR,
     )
 
-    if has_assignment(auth_client.role_assignments.list_for_scope(vnet), role_definition_id, object_id):
-        return
-
     _create_role_assignment(auth_client, vnet, RoleAssignmentCreateParameters(
         role_definition_id=role_definition_id,
         principal_id=object_id,
         principal_type='ServicePrincipal',
     ))
 
 
-def assign_contributor_to_routetable(cli_ctx, subnets, object_id):
+def has_contributor_on_vnet(cli_ctx, vnet, object_id):
+    auth_client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_AUTHORIZATION)
+
+
+    role_definition_id = resource_id(
+        subscription=get_subscription_id(cli_ctx),
+        namespace='Microsoft.Authorization',
+        type='roleDefinitions',
+        name=NETWORK_CONTRIBUTOR,
+    )
+
+    return has_assignment(auth_client.role_assignments.list_for_scope(vnet),
+                          role_definition_id, object_id)
+
+
+def has_contributor_on_routetable(cli_ctx, route_table, object_id):
+    auth_client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_AUTHORIZATION)
+
+    role_definition_id = resource_id(
+        subscription=get_subscription_id(cli_ctx),
+        namespace='Microsoft.Authorization',
+        type='roleDefinitions',
+        name=NETWORK_CONTRIBUTOR,
+    )
+
+    return has_assignment(auth_client.role_assignments.list_for_scope(route_table),
+                          role_definition_id, object_id)
```

Comment:
Im not sure if we can do this (not a python guy), but these looks very similar. could this be something like:
```
def has_network_contributor_on_object(cli_ctx, object, object_id):
    auth_client = get_mgmt_service_client(cli_ctx, ResourceType.MGMT_AUTHORIZATION)

    role_definition_id = resource_id(
        subscription=get_subscription_id(cli_ctx),
        namespace='Microsoft.Authorization',
        type='roleDefinitions',
        name=NETWORK_CONTRIBUTOR,
    )

    return has_assignment(auth_client.role_assignments.list_for_scope(object),
                          role_definition_id, object_id)
```
?

Reply:
Updated per previous comment.  Used `resource` as object is keyword in python